plugin/plugin.go

```diff
@@ -10,18 +10,34 @@ import (
 	"github.com/open-policy-agent/opa-envoy-plugin/internal"
 )
 
-// Factory defines the interface OPA uses to instantiate a plugin.
-type Factory struct{}
+// AuthZFactory defines the factory for the AuthZ plugin.
+type AuthZFactory struct{}
 
-// PluginName is the name to register with the OPA plugin manager
-const PluginName = internal.PluginName
+// ExtProcFactory defines the factory for the ExtProc plugin.
+type ExtProcFactory struct{}
 
-// New returns the object initialized with a valid plugin configuration.
-func (Factory) New(m *plugins.Manager, config interface{}) plugins.Plugin {
-	return internal.New(m, config.(*internal.Config))
+// Plugin names to register with the OPA plugin manager.
+const (
+	AuthZPluginName   = "envoy_ext_authz_grpc"
+	ExtProcPluginName = "envoy_ext_proc_grpc"
+)
+
+// New method for AuthZFactory.
```

Comment:
So this change would be break backward compatibility, correct? The rest of the changes seem backward compatible. Is there a way to either keep the old one and if not we can deprecate it in favor of a more flexible implementation.

Reply:
I'm not sure to be honest, if we consider the changed structure It'll start the authz plugin with "envoy_ext_authz_grpc" but the wrapper for the v2 is still there, unless it requires the naming to be "envoy.ext_authz.grpc" then yes it'll break.

If it can use the same NewAuthZ internal method what we can do is something like this:

plugin.go
```go
type Factory struct{}

func (Factory) New(m *plugins.Manager, config interface{}) plugins.Plugin {
	return internal.NewAuthZ(m, config.(*internal.Config))
}

func (Factory) Validate(m *plugins.Manager, configBytes []byte) (interface{}, error) {
	return internal.Validate(m, configBytes)
}
``` 

and for the main.go we move back to use the Factory
```go
runtime.RegisterPlugin("envoy.ext_authz.grpc", plugin.Factory{})
``` 

What do you think? I'm not sure how to test it, so if you can provide me with some guidance we can check if it works

Reply:
Seems like it worked, I'll adjust the logs and work on the tests for the new files, let me know if you have any other concerns about the backward compatibility

Reply:
The rest seemed ok to me. Thanks for working on this!